### PR TITLE
add binlog usage restrictions

### PR DESCRIPTION
**Pull request by IANTHEREAL**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
Increase the restriction of using binlog in version 5.0

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- Other reference link(s): 
  - Async Commit: https://github.com/pingcap/docs-cn/pull/5607/files
  - Clustered Index: https://github.com/pingcap/docs-cn/pull/5771/files

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-binlog/tidb-binlog-overview.md [hunk 1]
```diff
@@ -45,7 +49,15 @@ TiDB Binlog 集群主要分为 Pump 和 Drainer 两个组件，以及 binlogctl
 
 ## 注意事项
 
-* 需要使用 TiDB v2.0.8-binlog、v2.1.0-rc.5 及以上版本，否则不兼容该版本的 TiDB Binlog。
+* TiDB Binlog 与 TiDB v5.0.0-rc 版本开始引入的以下特性不兼容，无法一起使用：
+
+  - [TiDB Clustered index 特性](/clustered-indexes.md#限制)：开启 TiDB Binlog 后 TiDB 不允许创建非单个整数列作为主键的聚簇索引；已创建的聚簇索引表的数据插入、删除和更新动作不会通过 TiDB Binlog 同步到下游。如需同步聚簇索引表，请使用 [TiCDC](/ticdc/ticdc-overview.md)；
+  - TiDB 系统变量 [tidb_enable_async_commit](/system-variables.md#tidb_enable_async_commit-从-v500-rc-版本开始引入)：启用 TiDB Binlog 后，开启该选项无法获得性能提升。要获得性能提升，建议使用 [TiCDC](/ticdc/ticdc-overview.md) 替代 TiDB Binlog。
+  - TiDB 系统变量 [tidb_enable_1pc](/system-variables.md#tidb_enable_1pc-从-v500-rc-版本开始引入)：启用 TiDB Binlog 后，开启该选项无法获得性能提升。要获得性能提升，建议使用 [TiCDC](/ticdc/ticdc-overview.md) 替代 TiDB Binlog。
```
Comment (leoppro) [inline tidb-binlog/tidb-binlog-overview.md:56]:
```suggestion
    - [TiDB Clustered index 特性](/clustered-indexes.md#限制)：开启 TiDB Binlog 后 TiDB 不允许创建非单个整数列作为主键的聚簇索引；已创建的聚簇索引表的数据插入、删除和更新动作不会通过 TiDB Binlog 同步到下游。如需同步聚簇索引表，请使用 [TiCDC](/ticdc/ticdc-overview.md)；
    - TiDB 系统变量 [tidb_enable_async_commit](/system-variables.md#tidb_enable_async_commit-从-v500-rc-版本开始引入)：启用 TiDB Binlog 后，开启该选项无法获得性能提升。要获得性能提升，建议使用 [TiCDC](/ticdc/ticdc-overview.md) 替代 TiDB Binlog。
    - TiDB 系统变量 [tidb_enable_1pc](/system-variables.md#tidb_enable_1pc-从-v500-rc-版本开始引入)：启用 TiDB Binlog 后，开启该选项无法获得性能提升。要获得性能提升，建议使用 [TiCDC](/ticdc/ticdc-overview.md) 替代 TiDB Binlog。
```

File: tidb-binlog/tidb-binlog-overview.md [hunk 2]
```diff
@@ -45,7 +49,15 @@ TiDB Binlog 集群主要分为 Pump 和 Drainer 两个组件，以及 binlogctl
...
+
+* TiDB Binlog 与 TiDB v4.0.7 版本开始引入的以下特性不兼容，无法一起使用：
+
+  - TiDB 系统变量 [tidb_enable_amend_pessimistic_txn](/system-variables.md#tidb_enable_amend_pessimistic_txn-从-v407-版本开始引入)：两个功能存在兼容性问题，一起使用会造成 TiDB Binlog 复制数据不一致的正确性问题。
```
Comment (leoppro) [inline tidb-binlog/tidb-binlog-overview.md:60]:
```suggestion
    - TiDB 系统变量 [tidb_enable_amend_pessimistic_txn](/system-variables.md#tidb_enable_amend_pessimistic_txn-从-v407-版本开始引入)：两个功能存在兼容性问题，一起使用会造成 TiDB Binlog 复制数据不一致的正确性问题。
```

**Comment by IANTHEREAL** [comment]
@sticnarf PTAL

**Review by leoppro — COMMENTED** [review]
/lgtm

**Review by sticnarf — APPROVED** [review]
LGTM

File: tidb-binlog/tidb-binlog-overview.md [hunk 3]
```diff
@@ -45,7 +49,15 @@ TiDB Binlog 集群主要分为 Pump 和 Drainer 两个组件，以及 binlogctl
 
 ## 注意事项
 
-* 需要使用 TiDB v2.0.8-binlog、v2.1.0-rc.5 及以上版本，否则不兼容该版本的 TiDB Binlog。
+* TiDB Binlog 与 TiDB v5.0.0-rc 版本开始引入的以下特性不兼容，无法一起使用：
+
+    - [TiDB Clustered index 特性](/clustered-indexes.md#限制)：开启 TiDB Binlog 后 TiDB 不允许创建非单个整数列作为主键的聚簇索引；已创建的聚簇索引表的数据插入、删除和更新动作不会通过 TiDB Binlog 同步到下游。如需同步聚簇索引表，请使用 [TiCDC](/ticdc/ticdc-overview.md)；
```
Comment (yikeke) [inline tidb-binlog/tidb-binlog-overview.md:54]:
```suggestion
    - [TiDB 聚簇索引特性](/clustered-indexes.md#限制)：开启 TiDB Binlog 后 TiDB 不允许创建非单个整数列作为主键的聚簇索引；已创建的聚簇索引表的数据插入、删除和更新动作不会通过 TiDB Binlog 同步到下游。如需同步聚簇索引表，请使用 [TiCDC](/ticdc/ticdc-overview.md)；
```

File: tidb-binlog/tidb-binlog-overview.md [hunk 4]
```diff
@@ -45,7 +49,15 @@ TiDB Binlog 集群主要分为 Pump 和 Drainer 两个组件，以及 binlogctl
...
+    - TiDB 系统变量 [tidb_enable_async_commit](/system-variables.md#tidb_enable_async_commit-从-v500-rc-版本开始引入)：启用 TiDB Binlog 后，开启该选项无法获得性能提升。要获得性能提升，建议使用 [TiCDC](/ticdc/ticdc-overview.md) 替代 TiDB Binlog。
+    - TiDB 系统变量 [tidb_enable_1pc](/system-variables.md#tidb_enable_1pc-从-v500-rc-版本开始引入)：启用 TiDB Binlog 后，开启该选项无法获得性能提升。要获得性能提升，建议使用 [TiCDC](/ticdc/ticdc-overview.md) 替代 TiDB Binlog。
+
+* TiDB Binlog 与 TiDB v4.0.7 版本开始引入的以下特性不兼容，无法一起使用：
+
+    - TiDB 系统变量 [tidb_enable_amend_pessimistic_txn](/system-variables.md#tidb_enable_amend_pessimistic_txn-从-v407-版本开始引入)：两个功能存在兼容性问题，一起使用会造成 TiDB Binlog 复制数据不一致的正确性问题。
```
Comment (yikeke) [inline tidb-binlog/tidb-binlog-overview.md:60]: 这个提醒得 cherry-pick 到 4.0 文档吧？

Comment (IANTHEREAL) [reply tidb-binlog/tidb-binlog-overview.md:60]: 可以，这个合并后，我把这段挪过去

**Review by yikeke — APPROVED** [review]

**Pull request merged**